python/dolfinx/wrappers/mesh.cpp

```diff
@@ -107,7 +107,7 @@ void declare_meshtags(py::module& m, std::string type)
           }))
       .def_property_readonly("dtype", [](const dolfinx::mesh::MeshTags<T>& self)
                              { return py::dtype::of<T>(); })
-      .def_readwrite("name", &dolfinx::mesh::MeshTags<T>::name)
+      .def_readwrite("_name", &dolfinx::mesh::MeshTags<T>::name)
```

Comment:
You cannot have the same name for a property exposed to Python that has a getter and setter. This would be the natural way to expose variables in a pythonic way. The main problem is that the `MeshTagMetaClass` is not used for all constructors of meshtags in the Python layer, thus making it very hard to use typing and mypy for external libraries.

Reply:
One option instead of retyping would be to automatically generate stubs (pyi files) for our pybind11 module:

https://mypy.readthedocs.io/en/stable/stubgen.html

or

https://github.com/sizmailov/pybind11-stubgen

However, I couldn't get it to work, admittedly only a five minute try.

Then (maybe) we would not need to re-wrap everything at the

Reply:
``stubgen -p cpp`` in the same location of ``cpp.so`` should do it.

Reply:
The main issue  is that the directory of `cpp.so` depends on the installation process (and should preferably be executed when installing the pybind layer in `setup.py`).

Reply:
Usually those tools produce draft stubs which often require some manual updates, I thought those would be committed. Creating them on-the-fly in `setup.py` is also possible, but it would introduce dependencies. Additionally, if the package is already installed ``stubgen -p dolfinx.cpp`` will find it.

Reply:
@jorgensd Do you not think that the 'fix' then would be to use the Pythonic constructors for MeshTags throughout?

Reply:
I think that would be the correct way of fixing it. It would mean that we need to wrap up `xdmf.read_meshtags`
and `meshtags_from_entities`. I guess we would need to use the C++ copy constructor to transfer the data from C++ to Python, or do you have any other suggestion as to how we could do this in a nice way?